### ISSUE TITLE
Move "Radius" from NavigationMesh to NavigationAgent to allow agents of different sizes to use the same NavigationArea

### ISSUE THREAD

**Issue by matthew798** [godotengine/godot-proposals#5977]
### Describe the project you are working on

A generic 3d top-down strategy game

### Describe the problem or limitation you are having in your project

Currently, when baking a `NavigationMesh`, one must specify `Agents -> Radius` in order to account for the "width" of the `NavigationAgents`. This method requires multiple `NavigationMesh` instances (and thus multiple `NavigationRegion3D` instances) when agents with varying radii are used. It is extremely cumbersome to update many `NavigationMesh`s when changing things. The more "agents" of varying radii, the more complicated it becomes to maintain.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

I propose that the `Agents -> Radius` setting be moved from the `NavigationMesh` to the `NavigationAgent`. This would greatly simplify the management of agents of varying radii, and seems more intuitive.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

Self-Explanatory

### If this enhancement will not be used often, can it be worked around with a few lines of script?

This enhancement will simplify a very important feature of the engine. It should probably not be an optional add-on,

### Is there a reason why this should be core and not an add-on in the asset library?

See-above

**Comment by smix8** [comment]
See https://github.com/godotengine/godot-proposals/issues/5542 that is worked on and should solve this in a different way.

The `NavigationAgent` already has a `radius` property but currently it is only used for the avoidance.

Apart from being to costly a NavigationMesh resource can't be used to derive other agent sizes at runtime with magic cause it has no information on the source geometry. All the NavigationMesh properties are just for the baking process with the NavigationMeshGenerator singleton that does all the parsing work and updates the NavigationMesh.

**Comment by matthew798** [comment]
@smix8 I read through the other proposal, it's interesting but I think perhaps I didn't explain myself clearly.

I'm not suggesting that the NavigationMesh needs to be aware of the size of the agents.

I am however making a few assumptions:
1. The Navigation Server has access to each agent's properties
2. The calculations for the next position, etc are all done in the Navigation Server class
3. When we move an agent, we want it to be _entirely_ within the Navigation Mesh

For example, what I'm proposing is to essentially always bake `NavigationMesh`es with `agent -> radius` = 0, and use each individual agent's radius in the path calculations instead. (I did mention using the `NavigationMesh` radius as an extra "padding")

In other words, the `NavigationMesh` _always_ extends _right up to_ the threshold of any obstacles. The `NavigationServer` simply adds the agent's radius into the mix when calculating a path.

**Comment by lawnjelly** [comment]
> In other words, the `NavigationMesh` _always_ extends _right up to_ the threshold of any obstacles. The `NavigationServer` simply uses the agent's radius to offset the path from the edges of the `NavigationMesh`

There's a lot of problems / edge cases with this in practice, which is why every navmesh implementation I've personally used simply bakes different agent sizes instead.

Think about what happens when an agent is too big to fit in an opening. Pushing the planes of the navmesh edge can push the agent into a different room etc. When pathfinding with A*, how do you know whether an agent can fit through a gap etc.
